src/components/molecules/DepartmentManagement/DepartmentManagement.tsx

```diff
@@ -0,0 +1,60 @@
+import { FC, useCallback } from 'react'
+import { useTranslation } from 'react-i18next'
+import {
+  useDepartmentsFetch,
+  useParallelMutationDepartment,
+} from 'hooks/requests/useDepartments'
+
+import EditableListContainer from 'components/molecules/EditableListContainer/EditableListContainer'
+import { ModalTypes, showModal } from 'components/organisms/modals/ModalRoot'
+import { EditDataType } from 'components/organisms/modals/EditableListModal/EditableListModal'
+import { showNotification } from 'components/organisms/NotificationRoot/NotificationRoot'
+import { NotificationTypes } from '../Notification/Notification'
+import { includes } from 'lodash'
+
+import useAuth from 'hooks/useAuth'
+import { Privileges } from 'types/privileges'
+
+const DepartmentManagement: FC = () => {
+  const { t } = useTranslation()
+  const { userPrivileges } = useAuth()
+  const { existingDepartments } = useDepartmentsFetch()
+  const { parallelUpdating, isLoading } = useParallelMutationDepartment()
+
+  const handleOnSubmit = useCallback(
+    async (values: EditDataType[]) => {
+      await parallelUpdating(values)
+      showNotification({
+        type: NotificationTypes.Success,
+        title: t('notification.announcement'),
+        content: t('success.department_updated'),
+      })
+    },
+    [parallelUpdating, t]
+  )
+
+  const handleEditDepartmentsModal = () => {
+    showModal(ModalTypes.EditableListModal, {
+      editableData: existingDepartments,
+      title: t('modal.edit_departments'),
+      handleOnSubmit: handleOnSubmit,
+      isLoading,
```

Comment:
**[blocker]**

Didn't test, but I have a feeling that this "isLoading" value doesn't work. It will be "false" when the button is clicked and modal is opened. Even if it turns to "true" while the modal is open, the modal won't get that update, unless it is opened again with handleEditDepartmentsModal